File: Algorithms/Binary Tree/BinaryTree.py
class node(object):
    def __init__(self,value):
        self.value = value
        self.left = None
        self.right = None
class BinaryTree(object):
    def __init__(self,root):
        self.root = node(root)
    def print_tree(self,strr):
        if strr=='preorder':
            return self.preorder_print(tree.root,"")
        if strr =='inorder':
            return self.inorder_print(tree.root,"")
        if strr =='postorder':
            return self.postorder_print(tree.root,"")
    def preorder_print(self,start,traversal):
        if start:
            traversal += (str(start.value)+'-')
            traversal = self.preorder_print(start.left,traversal)
            traversal = self.preorder_print(start.right,traversal)
        return traversal
    def inorder_print(self,start,traversal):
        if start:
            traversal = self.inorder_print(start.left,traversal)
            traversal+=str(start.value) + '-'
            traversal = self.inorder_print(start.right,traversal)
        return traversal
    def postorder_print(self,start,traversal):
        if start:
            traversal = self.preorder_print(start.left,traversal)
            traversal = self.preorder_print(start.right,traversal)
            traversal += (str(start.value)+'-')
        return traversal
tree = BinaryTree(1)
tree.root.left = node(2)
tree.root.left.left = node(4)
tree.root.left.right = node(5)
tree.root.right = node(3)
tree.root.right.left = node(6)
tree.root.right.right = node(7)
print(tree.print_tree('preorder'))
print(tree.print_tree('inorder'))
print(tree.print_tree('postorder'))
#        1
 #      /  \
 #     2     3
 #    / \   / \
 #   4   5  6  7

File: Algorithms/binarytree.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int key;
    struct node *left, *right;
};
struct node *newNode(int item)
{
    struct node *temp =  (struct node *)malloc(sizeof(struct node));
    temp->key = item;
    temp->left = temp->right = NULL;
    return temp;
}
void inorder(struct node *root)
{
    if (root != NULL)
    {
        inorder(root->left);
        printf("%d \t", root->key);
        inorder(root->right);
    }
}

void preorder(struct node *root)
{
 if(root!=NULL)
 {
  printf(" %d \t",root->key);
  preorder(root->left);
  preorder(root->right);
 }
}

void postorder(struct node *root)
{
 if(root!=NULL)
 {
  postorder(root->left);
  postorder(root->right);
  printf(" %d \t",root->key);
 }
}

struct node* insert(struct node* node, int key)
{
    if (node == NULL) return newNode(key);
    if (key > node->key)
        node->left  = insert(node->left, key);
    else if (key < node->key)
        node->right = insert(node->right, key);
    return node;
}
int main()
{
    struct node *root = NULL;
    int ch,k;
    printf("1 Insertion \n");
    printf("2 Inorder \n");
    printf("3 Preorder \n");
    printf("4 Postorder \n");
    printf("5 - Exit\n");

    while(1)
    {
        printf("\nEnter your choice : ");
        scanf("%d", &ch);
        switch (ch)
        {
        case 1:
            {
             printf("enter the element t\t\t");
             scanf("%d",&k);
             root = insert(root, k);
             break;
            }

        case 2:{printf("inorder traversed  \n");
            inorder(root);
            break;}
        case 3:{printf("preorder traversed  \n");
            preorder(root);
            break;}
        case 4:{printf("postorder  \n");
            postorder(root);
            break;}

        case 5:
            exit(0);

        default :printf("Wrong choice, Please enter correct choice  ");
        }
    }
    return 0;
}
